File: bintree_py/test/testclone01.py
from .testbase import display_tree
from binarytree import AvlTree


def do_task():
    a = AvlTree()

    for value in [10, 20, 30, 40, 50, 25, 100, 28, 140]:
        a.insert(value)

    print(' display tree a:')
    display_tree(a)

    ###########################################

    b = a.clone()
    b.insert(22)
    b.insert(29)
    print('\n\n\n display tree b:')
    display_tree(b)
    print('\n\n height tree b:', b.height())

    ###########################################

    print('\n\n\n display tree a (again):')
    display_tree(a)

    ###########################################

    c = a.clone()
    c.clear()
    print('\n\n\n display tree c:')
    display_tree(c)

    print()

File: bintree_java/src/test/TestClone01.java
package test;


import binarytree.AvlTree;



public class TestClone01 extends TestBase {

    @Override
    public void doTask() {
        var a = new AvlTree<Integer>();

        for ( var value : new int[] {10, 20, 30, 40, 50, 25, 100, 28, 140} )
            a.insert(value);


        System.out.println(" display tree a:");
        displayTree(a);


        var b = a.clone();
        b.insert(22);
        b.insert(29);
        System.out.println("\n\n\n display tree b:");
        displayTree(b);
        System.out.println("\n\n height tree b: " + b.height());


        System.out.println("\n\n\n display tree a (again):");
        displayTree(a);


        var c = a.clone();
        c.clear();
        System.out.println("\n\n\n display tree c:");
        displayTree(c);


        System.out.println();
    }

}
